src/models.py

```diff
@@ -69,6 +69,21 @@ def build_model(args, vocab, pretrained_embs, tasks):
         cove_layer = None
         embedder = OpenAIEmbedderModule(args)
         d_emb = embedder.get_output_dim()
+    elif args.bert_model_name:
+        # Note: incompatible with other embedders, but logic in preprocess.py
+        # should prevent these from being enabled anyway.
+        from .bert.utils import BertEmbedderModule
+        log.info(f"Using BERT model ({args.bert_model_name}); skipping other embedders.")
+        cove_layer = None
+        # Set PYTORCH_PRETRAINED_BERT_CACHE environment variable to an existing
+        # cache; see
+        # https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/pytorch_pretrained_bert/file_utils.py
+        bert_cache_dir = (os.path.join(args.exp_dir, "bert_cache") if
+                          args.bert_fine_tune else None)
+        if bert_cache_dir:
```

Comment:
dumb question: why only cache if fine tuning? Is the cache dir for storing fine tuned weights?

Reply:
Good point; I think this is unnecessary. Safest thing is to download once then mark as read-only, I guess?

There was a line in the documentation advising to use separate directories for each worker in distributed training, but this might just be to avoid hammering a single file server.

Reply:
Removed this special case; should always use cache if the environment var is set.